docs/autodiff.rst

```diff
@@ -0,0 +1,239 @@
+.. _differentiable:
+
+Differentiable programming
+==========================
+
+We suggest starting with the ``ti.Tape()``, and then migrate to more advanced differentiable programming using the ``kernel.grad()`` syntax if necessary.
+
+
+Introduction
+------------
+
+For example, you have the following kernel:
+
+.. code-block:: python
+
+    x = ti.var(ti.f32, ())
+    y = ti.var(ti.f32, ())
+
+    @ti.kernel
+    def compute_y():
+        y[None] = ti.sin(x[None])
+
+
+Now you want to get the derivative of y corresponding to x, i.e. dy/dx.
+You may want to write the derivatived kernel by hand:
```

Comment:
```suggestion
Now if you want to get the derivative of y corresponding to x, i.e., dy/dx.
You may want to implement the derivative kernel by yourself:
```
There should be a comma behind i.e. right? It's the abbreviation. And using the adj of derivative should be enough? wdyt :)